benchmark/httpapi.py

```diff
@@ -100,6 +112,104 @@ def delete(self, id):
             return fail(ResultNotFound())
 
 
+@implementer(IBackend)
+class TxMongoBackend(object):
+    """
+    The backend that uses txmongo driver to work with MongoDB.
+    """
+    def __init__(self, hostname="127.0.0.1", port=27017):
+        connection = MongoConnectionPool(host=hostname, port=port)
+        self.collection = connection.benchmark.results
+
+    def disconnect(self):
+        return self.collection.database.connection.disconnect()
+
+    def store(self, result):
+        """
+        Store a single benchmarking result and return its identifier.
+
+        :param dict result: The result in the JSON compatible format.
+        :return: A Deferred that produces an identifier for the stored
+            result.
+        """
+        def to_str(result):
+            return str(result.inserted_id)
+
+        # Store the timestamp field as a special hidden datetime field
+        # for sorting.
+        result['sort$timestamp'] = timestamp_parser.parse(result['timestamp'])
+        id = self.collection.insert_one(result)
+        id.addCallback(to_str)
+        return id
+
+    def retrieve(self, id):
+        """
+        Retrive a result by the given identifier.
+        """
+        try:
+            object_id = ObjectId(id)
+        except InvalidId:
+            raise ResultNotFound()
```

Comment:
If it's an InvalidId then it's probably a programming error as we shouldn't
be called with an Id that we didn't generate? I think that letting the InvalidId
exception propagate would avoid masking that error.